### DISCUSSION TITLE
Button (or Console Command) to manually save Total-Energy through Web-UI (before unpluigging)

### DISCUSSION THREAD

**Discussion by tarator** [arendst/Tasmota#19140] [Ideas]
A simple solution to prevent Total energy of the day to get lost when unplugging Tasmota Power plug (see [Ticket #9132](https://github.com/arendst/Tasmota/issues/9132)  would be to implement a Button where a user can save this data.

Or is there already a command which can be issued in the Console?

**Comment by Jason2866** [comment]
Yes, there is `restart 2` which shuts down Tasmota correctly. Needs a hardware reset or a power cycle to start Tasmota.

**Reply by tarator** [reply]
Sorry for that noob question:

> Needs a hardware reset or a power cycle to start Tasmota.

you mean: unplugging and then plugging in again?

**Reply by Jason2866** [reply]
Yes

**Reply by sfromis** [reply]
Additionally, just doing a normal `Restart 1` or using the Restart button in the web UI should also save settings, including energy consumption data.
